calliope/test/test_constraint_results.py

```diff
@@ -799,3 +799,46 @@ def test_location_specific_energy_cap_min_constraint(self):
         expensive_capacity1 = capacity.loc[("1", "expensive_supply")]
         assert round(expensive_capacity0, 5) >= 6
         assert expensive_capacity1 == 0
+
+
+class TestStorageCapacityRatio:
+
+    def test_no_ratio_set(self):
```

Comment:
I would recommend testing these without actually running the optimisation. See the tests in `test_backend_pyomo` which check if a particular variable exists in a constraint (e.g. line 214). In this case, you could check that the storage capacity constraints include the variable `energy_cap`, which they only will if your constraint has been applied